usaspending_api/accounts/views/financial_spending.py

```diff
@@ -44,5 +45,46 @@ def get_queryset(self):
         # sum obligated_mount by object class
         queryset = queryset.values('major_object_class_name', 'major_object_class_code').annotate(
             obligated_amount=Sum('obligations_incurred_by_program_object_class_cpe'))
+        # get minor object class vars
+        return queryset
+
+
+class MinorObjectClassFinancialSpendingViewSet(DetailViewSet):
+    """Returns financial spending data by object class."""
+
+    serializer_class = MinorObjectClassFinancialSpendingSerializer
+
+    def get_queryset(self):
+        # retrieve post request payload
+        json_request = self.request.query_params
+
+        # retrieve fiscal_year & agency_id from request
+        fiscal_year = json_request.get('fiscal_year', None)
+        funding_agency_id = json_request.get('funding_agency_id', None)
+        major_object_class_code = json_request.get('major_object_class_code', None)
+
+        # required query parameters were not provided
+        if not (fiscal_year and funding_agency_id and major_object_class_code):
+            raise InvalidParameterException('Missing one or more required query parameters: fiscal_year, funding_agency_id')
 
+        # using final_objects below ensures that we're only pulling the latest
+        # set of financial information for each fiscal year
+        queryset = FinancialAccountsByProgramActivityObjectClass.final_objects.all()
+        # get the incoming agency's toptier agency, because that's what we'll
+        # need to filter on
+        # (used filter() instead of get() b/c we likely don't want to raise an
+        # error on a bad agency id)
+        toptier_agency = Agency.objects.filter(id=funding_agency_id).first().toptier_agency
```

Comment:
line 24 checks to make sure the value is provided (if not provided an invalad parameter exception is thrown).  if the provided id doesnt exist, an empty results array is returned.  This is the behavior in other endpoints.